xdsl/dialects/onnx.py

```diff
@@ -673,6 +675,94 @@ def verify_(self) -> None:
                 f"Only one value attribute must be provided, but {used_attrs} were specified"
             )
 
+    def get_value(self) -> tuple[Any | None, str]:
+        if self.value_float:
+            return self.value_float, "value_float"
+        elif self.value_int:
+            return self.value_int, "value_int"
+        elif self.value_ints:
+            # In the ONNX-MLIR format, integer types are abstracted, so only the integer data is extracted
+            # for printing purposes.
+            arr: list[int] = [value.value.data for value in self.value_ints]
+            return arr, "value_ints"
+        else:
+            # TODO: value_string, value_strings, value_floats
+            raise NotImplementedError()
+
```

Comment:
Sorry to change my advice but, thinking about this more, I think we might as well do it a different way. Can you please change this so that it returns an Attribute and a string, not an optional Any? I don't think that it makes sense to add logic that's specific to printing outside of the printing method

Reply:
I agree, it would also avoid the duplication a bit